File: services/api/app/schemas/pseudonymous.py
"""Schemas: pseudonyme Anmeldung über eine Fachperson.

Eine eingeladene Person legt mit Pseudonym + Passwort (ohne echte E-Mail) ein
Konto an. Siehe services/pseudonymous_service.py.
"""
from __future__ import annotations

from pydantic import BaseModel, Field


class PseudonymousRegisterRequest(BaseModel):
    token: str | None = None
    code: str | None = None
    handle: str = Field(..., min_length=3, max_length=30)
    password: str = Field(..., min_length=8, max_length=200)


class PseudonymousRegisterResponse(BaseModel):
    login_email: str
    recovery_code: str
    professional_display_name: str | None = None


class PseudonymousRecoverRequest(BaseModel):
    handle: str = Field(..., min_length=1, max_length=60)
    recovery_code: str = Field(..., min_length=1, max_length=60)
    new_password: str = Field(..., min_length=8, max_length=200)


class PseudonymousRecoverResponse(BaseModel):
    login_email: str
    recovery_code: str

File: apps/web/src/api/pseudonymous.ts
import { apiClient } from './client'

/** Pseudonyme Anmeldung über eine Fachperson (kein Klarname bei EchoB). */
export interface PseudonymousRegisterResult {
  login_email: string
  recovery_code: string
  professional_display_name: string | null
}

export interface PseudonymousRecoverResult {
  login_email: string
  recovery_code: string
}

export const pseudonymousApi = {
  /** Legt ein pseudonymes Konto an (an eine gültige Einladung gebunden). */
  register: (data: { token?: string; code?: string; handle: string; password: string }) =>
    apiClient.post<PseudonymousRegisterResult>('/pseudonymous/register', data).then(r => r.data),

  /** Setzt das Passwort über den Wiederherstellungs-Code zurück (rotiert ihn). */
  recover: (data: { handle: string; recovery_code: string; new_password: string }) =>
    apiClient.post<PseudonymousRecoverResult>('/pseudonymous/recover', data).then(r => r.data),
}
